vicon/udp.h

```diff
@@ -39,98 +39,100 @@ namespace BoBRobotics {
+    void update(uint32_t frameNumber, millimeter_t x, millimeter_t y, millimeter_t z,
+                radian_t yaw, radian_t pitch, radian_t roll)
     {
         // Cache frame number
         m_FrameNumber = frameNumber;
 
         // Copy vectors into class
-        m_Position[0] = x; m_Position[1] = y; m_Position[2] = z;
-        m_Attitude[0] = yaw; m_Attitude[1] = pitch; m_Attitude[2] = roll;
+        m_Position[0] = x;
+        m_Position[1] = y;
+        m_Position[2] = z;
+        m_Attitude[0] = yaw;
+        m_Attitude[1] = pitch;
+        m_Attitude[2] = roll;
+    }
+
+    uint32_t getFrameNumber() const
+    {
+        return m_FrameNumber;
+    }
+
+    template <class LengthUnit = millimeter_t>
+    Triple<LengthUnit> getPosition()
+    {
+        return makeUnitTriple<LengthUnit>(m_Position);
     }
 
-    uint32_t getFrameNumber() const { return m_FrameNumber; }
-    Vector<LengthUnit> &getPosition() override { return m_Position; }
-    Vector<AngleUnit> &getAttitude() override { return m_Attitude; }
+    template <class AngleUnit = radian_t>
+    Triple<radian_t> getAttitude()
+    {
+        return makeUnitTriple<AngleUnit>(m_Attitude);
+    }
 
 private:
     //----------------------------------------------------------------------------
     // Members
     //----------------------------------------------------------------------------
     uint32_t m_FrameNumber;
-    Vector<LengthUnit> m_Position;
-    Vector<AngleUnit> m_Attitude;
+    Array3<millimeter_t> m_Position;
+    Array3<radian_t> m_Attitude;
 };
 
 //----------------------------------------------------------------------------
 // Vicon::ObjectDataVelocity
 //----------------------------------------------------------------------------
 //! Object data class which also calculate (un-filtered) velocity
-template <class LengthUnit = meter_t, class AngleUnit = radian_t, class VelocityUnit = meters_per_second_t>
-class ObjectDataVelocity : public ObjectData<LengthUnit, AngleUnit>
+class ObjectDataVelocity : public ObjectData
 {
-static_assert(units::traits::is_velocity_unit<VelocityUnit>::value,
-              "VelocityUnit must be a velocity type (e.g. meters_per_second_t)");
-
 public:
     ObjectDataVelocity() : m_Velocity{0_mps, 0_mps, 0_mps}
-    {
-    }
+    {}
 
     //----------------------------------------------------------------------------
     // Public API
     //----------------------------------------------------------------------------
-    void update(uint32_t frameNumber, LengthUnit x, LengthUnit y, LengthUnit z,
-                AngleUnit yaw, AngleUnit pitch, AngleUnit roll)
+    void update(uint32_t frameNumber, millimeter_t x, millimeter_t y, millimeter_t z,
+                radian_t yaw, radian_t pitch, radian_t roll)
     {
-        const Vector<LengthUnit> position {x, y, z};
-        constexpr second_t frameS = 10_ms;
-        constexpr second_t smoothingS = 30_ms;
+        const Triple<millimeter_t> position {x, y, z};
```

Comment:
Why are you turning this into a tuple? All you do with it is extract it's x, y and z a few lines down...